tests/unit/test_utils.py

```diff
@@ -628,21 +628,50 @@ def test_call_subprocess_closes_stdin():
 
 
 @pytest.mark.parametrize('auth_url, expected_url', [
-    ('https://user:pass@domain.tld/project/tags/v0.2',
-     'https://domain.tld/project/tags/v0.2'),
     ('https://domain.tld/project/tags/v0.2',
      'https://domain.tld/project/tags/v0.2',),
-    ('https://user:pass@domain.tld/svn/project/trunk@8181',
-     'https://domain.tld/svn/project/trunk@8181'),
     ('https://domain.tld/project/trunk@8181',
      'https://domain.tld/project/trunk@8181',),
+    ('https://user:pass@domain.tld/project/tags/v0.2',
+     'https://domain.tld/project/tags/v0.2'),
+    ('https://user:pass@domain.tld/svn/project/trunk@8181',
+     'https://domain.tld/svn/project/trunk@8181'),
     ('git+https://pypi.org/something',
      'git+https://pypi.org/something'),
-    ('git+https://user:pass@pypi.org/something',
-     'git+https://pypi.org/something'),
     ('git+ssh://git@pypi.org/something',
      'git+ssh://pypi.org/something'),
+    ('git+https://user:pass@pypi.org/something',
+     'git+https://pypi.org/something'),
 ])
 def test_remove_auth_from_url(auth_url, expected_url):
     url = remove_auth_from_url(auth_url)
     assert url == expected_url
+
+
+@pytest.mark.parametrize('auth_url, expected_url', [
+    ('http://user@domain.tld:8080/',
+     'http://user@domain.tld:8080/',),
```

Comment:
I don't see the test case I suggested of a netloc containing a `:` with no `@` for the netloc, like "domain.tld:8080", to make sure the characters following the colon aren't stripped.